### PR TITLE
#345  Gulp build in docker

### LINKED ISSUE #345: Do `gulp build` on image side
Currently, we have problems with gulp build on client side. See #344 for details.
Moreover, building static on container site is more good practice

### PR DESCRIPTION
**Pull request by duker33**
Closes #345

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -33,10 +33,13 @@ services:
...
     image: fidals/se-nodejs:dev
-    volumes_from:
-      - app
+    # For refarm-site's front development
+    # volumes:
+    #  - $REFARM_DIR:/usr/app/refarm-site
     env_file:
       - env_files/paths
+    environment:
+      - DEPS_DIR=/usr/app/refarm-site
```
Comment (artemiy312) [inline docker/docker-compose.yml]: We are storing a `DEPS_DIR` in `env_files/paths`, so this seems redundant

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -33,10 +33,13 @@ services:
 
   nodejs:
     image: fidals/se-nodejs:dev
-    volumes_from:
-      - app
+    # For refarm-site's front development
```
Comment (artemiy312) [inline docker/docker-compose.yml]: Why did you remove `volumes_from`? How will we perform `gulp watch/build` for local changes?

File: docker/env_files/paths.dist [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Identify the dependencies folder
 DEPS_DIR=/usr/local/lib/python3.6/site-packages
+REFARM_DIR=/path/to_my/refarm_site
```
Comment (artemiy312) [inline docker/env_files/paths.dist]: `REFARM_DIR` is redundant, because `DEPS_DIR` has the same function, isn't it?

File: docker/images/node/Dockerfile [hunk 4]
```diff
@@ -2,7 +2,19 @@ FROM node:slim
 
 WORKDIR /usr/app/src/
 
+ENV DEPS_DIR=/usr/app/refarm-site
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends --no-install-suggests -y ssh git \
+    && git clone https://github.com/fidals/refarm-site.git $DEPS_DIR \
```
Comment (artemiy312) [inline docker/images/node/Dockerfile]: The refarm-site is versioned, so we should get its version from requirements.txt or somewhere else

File: docker/docker-compose.yml (hunk 1 above)
Comment (duker33) [reply docker/docker-compose.yml]: Done.

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -33,8 +35,12 @@ services:
 
   nodejs:
     image: fidals/se-nodejs:dev
-    volumes_from:
-      - app
+    volumes:
+      - $FRONT_BUILD_DIR
+      # Volumes for refarm-site's front development
+      #- ../gulpfile.babel.js:/usr/app/src_front/gulpfile.babel.js
```
Comment (artemiy312) [inline docker/docker-compose.yml]: Perhaps we should mount these volumes by default. It seems more convenient, because docker-compose.yml is used to for development purpose

**Comment by artemiy312** [comment]
@duker33 npm [build failed](https://ci.fidals.com/fidals/shopelectro/140/7)

**Comment by duker33** [comment]
all passed except lint fixing at https://github.com/fidals/stroyprombeton/issues/200.
Merge it

File: docker/docker-compose.yml (hunk 5 above)
Comment (duker33) [reply docker/docker-compose.yml]: @artemiy312 , sorry, missed this comment.

I'll enable it buy default
